### PR TITLE
[Patch] DR-2013: Trim project labels to 63 characters

### PR DESCRIPTION
**Pull request by snf2ye**

### CODE REVIEW

File: src/test/java/bio/terra/service/resourcemanagement/google/GoogleProjectServiceTest.java [hunk 1]
```diff
@@ -85,4 +88,21 @@ public void testAppEngineOpIdExtraction() {
...
+    String trimmedName = GoogleProjectService.cleanForLabels(tooLongName);
+    assertEquals("Project label should be trimmed down when too long", trimmedName.length(), 63);
+
+    String nameWithCharacters = "workflow!_launcher+TESTING_dataset5243fe12db1640";
+    assertTrue(
+        "Original Project label should contain non-valid characters",
+        nameWithCharacters.contains("!"));
+    String cleanedName = GoogleProjectService.cleanForLabels(nameWithCharacters);
+    assertFalse(
+        "Original Project label should no longer contain non-valid characters",
```
Comment (nmalfroy) [inline src/test/java/bio/terra/service/resourcemanagement/google/GoogleProjectServiceTest.java:105]: Conversely, you could also compare explicitly to a cleaned string

Comment (nmalfroy) [reply src/test/java/bio/terra/service/resourcemanagement/google/GoogleProjectServiceTest.java:105]: That way you're testing the lowercasing, etc.

**Review by nmalfroy — APPROVED** [review]
thanks for fixing this!

**Comment by snf2ye** [comment]
Not sure if this was necessary, but I just created a new branch that's based on the latest version from production - 1.148.0. https://github.com/DataBiosphere/jade-data-repo/pull/1021

@nmalfroy - I think I've taken your comments into account there.

**Pull request closed without merging**